translator/translate/otel/processor/awsentity/translator.go

```diff
@@ -0,0 +1,94 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package awsentity
+
+import (
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/processor"
+
+	"github.com/aws/amazon-cloudwatch-agent/plugins/processors/awsentity"
+	"github.com/aws/amazon-cloudwatch-agent/translator/context"
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/util"
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/common"
+	"github.com/aws/amazon-cloudwatch-agent/translator/util/ecsutil"
+)
+
+const (
+	name     = "awsentity"
+	Service  = "Service"
+	Resource = "Resource"
+)
+
+type translator struct {
+	factory    processor.Factory
+	entityType string
+	name       string
+}
+
+func NewTranslator() common.Translator[component.Config] {
+	return &translator{
+		factory: awsentity.NewFactory(),
+	}
+}
+
+func NewTranslatorWithEntityType(entityType string) common.Translator[component.Config] {
+	return &translator{
+		factory:    awsentity.NewFactory(),
+		entityType: entityType,
+		name:       strings.ToLower(entityType),
+	}
+}
+
+func (t *translator) ID() component.ID {
+	return component.NewIDWithName(t.factory.Type(), t.name)
+}
+
+func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
+	// Do not send entity for ECS
+	if context.CurrentContext().RunInContainer() && ecsutil.GetECSUtilSingleton().IsECS() {
+		return nil, nil
+	}
```

Comment:
This should be checked in the pipeline instead. It's not an issue because all cases of this are already checked in the two pipelines that this was added to (`host`, `applicationsignals`).

Reply:
Blocking comment.